src/goto-programs/remove_exceptions.cpp

```diff
@@ -98,175 +98,114 @@ class remove_exceptionst
   std::map<irep_idt, std::set<irep_idt>> &exceptions_map;
   bool remove_added_instanceof;
 
-  void add_exceptional_returns(
-    const irep_idt &function_id,
-    goto_programt &goto_program);
+  symbol_exprt get_inflight_exception_global();
+
+  bool may_throw_or_receive_exceptions(const goto_programt &) const;
 
   void instrument_exception_handler(
-    const irep_idt &function_id,
     goto_programt &goto_program,
-    const goto_programt::targett &);
+    const goto_programt::targett &,
+    bool may_catch);
 
   void add_exception_dispatch_sequence(
-    const irep_idt &function_id,
     goto_programt &goto_program,
     const goto_programt::targett &instr_it,
     const stack_catcht &stack_catch,
     const std::vector<exprt> &locals);
 
   void instrument_throw(
-    const irep_idt &function_id,
     goto_programt &goto_program,
     const goto_programt::targett &,
     const stack_catcht &,
     const std::vector<exprt> &);
 
   void instrument_function_call(
-    const irep_idt &function_id,
     goto_programt &goto_program,
     const goto_programt::targett &,
     const stack_catcht &,
     const std::vector<exprt> &);
 
   void instrument_exceptions(
-    const irep_idt &function_id,
     goto_programt &goto_program);
 };
 
-/// adds exceptional return variables for every function that may escape
-/// exceptions
-void remove_exceptionst::add_exceptional_returns(
-  const irep_idt &function_id,
-  goto_programt &goto_program)
+/// Create a global named java::\@inflight_exception that holds any exception
+/// that has been thrown but not yet caught.
+symbol_exprt remove_exceptionst::get_inflight_exception_global()
 {
+  const symbolt *existing_symbol =
+    symbol_table.lookup(INFLIGHT_EXCEPTION_VARIABLE_NAME);
+  INVARIANT(
+    existing_symbol != nullptr,
```

Comment:
Nit: More canonical to just use `existing_symbol`?

Reply:
Nah, this makes it clearer what sort of operator-bool I'm looking for